### PR TITLE
Adds xk6-browser v1.0.2 release notes

### PR DESCRIPTION
**Pull request by ka3de**
## What?

Adds the release notes corresponding to xk6-browser `v1.0.2`.

### CODE REVIEW

File: release notes/v0.46.0.md
Comment (ankur22) [inline release notes/v0.46.0.md:110]:
```suggestion
- [browser#979](https://github.com/grafana/xk6-browser/pull/979) Temporary fix for a deadlock on `pagehide` event evaluation when `page.close()` is called.
```

File: release notes/v0.46.0.md
Comment (ankur22) [inline release notes/v0.46.0.md:145]: Would this better placed under improvements?

File: release notes/v0.46.0.md
Comment (ankur22) [inline release notes/v0.46.0.md:152]: Should this go in breaking changes?

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:110]: I'm not sure we should mention "temporary" in release notes, especially for something that is more internal. If the change fixes by now wouldn't it be better to just say "Fixes"?

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:145]: It is under improvements section, did you mean "enhancements" ?

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:152]: Yep you are right, it was an exported element after all. Let me fix it.

Comment (ka3de) [reply release notes/v0.46.0.md]: Fixed in https://github.com/grafana/k6/pull/3260/commits/86e3f0bfeb8b84eee51a5c3640de4e6ce73d0449.

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:51]: In the breaking section I added the entry for https://github.com/grafana/xk6-browser/pull/910, [here](https://github.com/grafana/k6/blob/86e3f0bfeb8b84eee51a5c3640de4e6ce73d0449/release%20notes/v0.46.0.md?plain=1#L17), which already explains the breaking changes in the API. But then I think we need to explain better what is the current API and how the browser lifecycle is handled, so I decided to include an entry in `New features` which, as per the format defined, seems to be the section that allows for further explanation, and include here grafana/xk6-browser#944 which, after the braking changes done in 910, it includes a new functionality to handle browser lifecycle based on k6 events. Does that make sense?

File: release notes/v0.46.0.md
Comment (mstoykov) [inline release notes/v0.46.0.md:11]: I wonder if this shouldn't be in the browser section below. And here just to be "Browser has a number of breaking changes due to change in vision on how it will be used, full list -> link"

Currently, The whole first page of the release notes is breaking changes ... and they are only in browser.

Anyone not interested in that will just gloss over them, potentially missing something (if we have other ones). 

Additionally reading this list will not really tell users they likely need to rewrite big parts of their scripts(or at least how to start browsers) - this happens in the other section. 

cc @grafana/k6-core

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md]: Fixed in [a7cf0a2](https://github.com/grafana/k6/pull/3260/commits/a7cf0a2710efff9f593b3f1e28e4e281df8e2a7a).

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:11]: Which browser section? You mean the _"Automatic Browser Lifecycle Handling"_ point in _"New Features"_ section? Or you mean adding a dedicated browser section at least for this release?

Comment (ka3de) [reply release notes/v0.46.0.md:11]: Personally I think providing the full list as it is now it's ok, but maybe we could add an H3 header just before the browser changes inside the _"Breaking changes"_ section with an introduction to the changes and links to the _"Automatic Browser Lifecycle Handling"_ point, which summarizes most of the important changes, similar to what you said I believe.

Our idea is to write a migration guide that will assist users in this transition, but unfortunately this is not yet ready so we can link it directly from the release notes.

File: release notes/v0.46.0.md
Comment (olegbespalov) [inline release notes/v0.46.0.md:146]: I'd recommend trying to squash things whenever this is possible. And also, try to see for the record first of all from the k6's user perspective (I mean, how valuable is the record for the user).

Like instead of:
* Refactored A
* Refactored B
* Added test C

It could be easily:
* Internal refactorings and test improvements (#A, #B, #C)

Comment (ka3de) [reply release notes/v0.46.0.md:146]: Personally I would remove many of these internal changes that do not impact the end user. How does this diff look to you?
cc: @ankur22 @inancgumus 

<details><summary>diff</summary>

```diff
diff --git a/release notes/v0.46.0.md b/release notes/v0.46.0.md
index 1f2528a9..d90d7cce 100644
--- a/release notes/v0.46.0.md  
+++ b/release notes/v0.46.0.md  
@@ -143,22 +143,12 @@ _Format as `<number> <present_verb> <object>. <credit>`_:
 - [#3136](https://github.com/grafana/k6/pull/3136) Updates k6 Go dependencies.
 - [#3131](https://github.com/grafana/k6/pull/3131) Updates our Pull Request template to be more structured and include a checklist for contributors.
 - [#3177](https://github.com/grafana/k6/pull/3177) Updates the version of golangci-lint we use to the latest version.
-- [browser#849](https://github.com/grafana/xk6-browser/pull/849) Updates `nil` session log warn to debug level in page attach.
-- [browser#889](https://github.com/grafana/xk6-browser/pull/889) Refactors `isRemoteBrowser` check.
-- [browser#889](https://github.com/grafana/xk6-browser/pull/899) Refactors pid registry.
-- [browser#902](https://github.com/grafana/xk6-browser/pull/902) Refactors registry back into browser package.
-- [browser#904](https://github.com/grafana/xk6-browser/pull/904) Fixes a race condition in Web Vitals unit test.
-- [browser#906](https://github.com/grafana/xk6-browser/pull/906) Parses the scenarios JSON definition from the `K6_INSTANCE_SCENARIOS` environment variable.
 - [browser#918](https://github.com/grafana/xk6-browser/pull/918) Replaces the usage of `os.LookupEnv` for k6 `LookupEnv` and abstracts environment variables lookup.
-- [browser#935](https://github.com/grafana/xk6-browser/pull/935) Refactors the integration test browser implementation.
-- [browser#936](https://github.com/grafana/xk6-browser/pull/936) Removes the `BaseEventEmitter` from browser.
-- [browser#959](https://github.com/grafana/xk6-browser/pull/959) Fixes `TestFrameNoPanicNavigateAndClickOnPageWithIFrames` test.
 - [browser#972](https://github.com/grafana/xk6-browser/pull/972) Fixes log level from `warn` to `debug` when `fetchBody` fails due to `context` being canceled on iteration end.
 - [browser#977](https://github.com/grafana/xk6-browser/pull/977) Removes goroutine ID field from logger.
 - [browser#978](https://github.com/grafana/xk6-browser/pull/978) Adds browser extension source log field.
 - [browser#983](https://github.com/grafana/xk6-browser/pull/983) Avoids `onRequestPaused` error log when `context` is canceled.
 - [browser#984](https://github.com/grafana/xk6-browser/pull/984) Ensures CDP requests message ID are unique at the connection scope.
-- [browser#992](https://github.com/grafana/xk6-browser/pull/992) Fixes go mod prometheus dependency issue.
```

</details>

File: release notes/v0.46.0.md
Comment (ankur22) [reply release notes/v0.46.0.md:51]: Yeah, I see what you've done, but as a user I think I'd like to know how to work with the new API and how to migrate to it. It is true that abstracting away some of the boiler plate is an improvement, but I wouldn't class it as a new feature either since it doesn't add anything new that a user couldn't do earlier.

We've also classed the browser module as a breaking change by increasing the major version, so I think that warrants a detailed explanation in the breaking changes section. I still feel that a single sub section under breaking changes where we have all the associated issues and an explanation would be a better place.

Comment (ka3de) [reply release notes/v0.46.0.md:51]: This does not seem to fit the format defined in the template, but is similar to my suggestion in this [comment](https://github.com/grafana/k6/pull/3260#discussion_r1289715563), so let's wait for k6 core team's feedback in that discussion.

Comment (ankur22) [reply release notes/v0.46.0.md:51]: True, but how do we best explain a major version change?

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:146]: I still think there are changes in the list that are irrelevant for end users, and grouping them just makes the other elements that fall in the same category but are more important to lose "weight/relevance" in the list.

Trying to follow your suggestions, how does this look?
<details><summary>diff</summary>

```diff
diff --git a/release notes/v0.46.0.md b/release notes/v0.46.0.md
index 1f2528a9..72438375 100644
--- a/release notes/v0.46.0.md  
+++ b/release notes/v0.46.0.md  
@@ -143,22 +143,12 @@ _Format as `<number> <present_verb> <object>. <credit>`_:
 - [#3136](https://github.com/grafana/k6/pull/3136) Updates k6 Go dependencies.
 - [#3131](https://github.com/grafana/k6/pull/3131) Updates our Pull Request template to be more structured and include a checklist for contributors.
 - [#3177](https://github.com/grafana/k6/pull/3177) Updates the version of golangci-lint we use to the latest version.
-- [browser#849](https://github.com/grafana/xk6-browser/pull/849) Updates `nil` session log warn to debug level in page attach.
-- [browser#889](https://github.com/grafana/xk6-browser/pull/889) Refactors `isRemoteBrowser` check.
-- [browser#889](https://github.com/grafana/xk6-browser/pull/899) Refactors pid registry.
-- [browser#902](https://github.com/grafana/xk6-browser/pull/902) Refactors registry back into browser package.
-- [browser#904](https://github.com/grafana/xk6-browser/pull/904) Fixes a race condition in Web Vitals unit test.
+- [browser#849](https://github.com/grafana/xk6-browser/pull/849), [browser#972](https://github.com/grafana/xk6-browser/pull/972), [browser#977](https://github.com/grafana/xk6-browser/pull/977), [browser#978](https://github.com/grafana/xk6-browser/pull/978), [browser#983](https://github.com/grafana/xk6-browser/pull/983) Log fixes and improvements.
+- [browser#889](https://github.com/grafana/xk6-browser/pull/889), [browser#889](https://github.com/grafana/xk6-browser/pull/899), [browser#902](https://github.com/grafana/xk6-browser/pull/902), [browser#935](https://github.com/grafana/xk6-browser/pull/935), [browser#936](https://github.com/grafana/xk6-browser/pull/936) Internal refactors.
+- [browser#904](https://github.com/grafana/xk6-browser/pull/904), [browser#959](https://github.com/grafana/xk6-browser/pull/959) Test fixes.
 - [browser#906](https://github.com/grafana/xk6-browser/pull/906) Parses the scenarios JSON definition from the `K6_INSTANCE_SCENARIOS` environment variable.
 - [browser#918](https://github.com/grafana/xk6-browser/pull/918) Replaces the usage of `os.LookupEnv` for k6 `LookupEnv` and abstracts environment variables lookup.
-- [browser#935](https://github.com/grafana/xk6-browser/pull/935) Refactors the integration test browser implementation.
-- [browser#936](https://github.com/grafana/xk6-browser/pull/936) Removes the `BaseEventEmitter` from browser.
-- [browser#959](https://github.com/grafana/xk6-browser/pull/959) Fixes `TestFrameNoPanicNavigateAndClickOnPageWithIFrames` test.
-- [browser#972](https://github.com/grafana/xk6-browser/pull/972) Fixes log level from `warn` to `debug` when `fetchBody` fails due to `context` being canceled on iteration end.
-- [browser#977](https://github.com/grafana/xk6-browser/pull/977) Removes goroutine ID field from logger.
-- [browser#978](https://github.com/grafana/xk6-browser/pull/978) Adds browser extension source log field.
-- [browser#983](https://github.com/grafana/xk6-browser/pull/983) Avoids `onRequestPaused` error log when `context` is canceled.
 - [browser#984](https://github.com/grafana/xk6-browser/pull/984) Ensures CDP requests message ID are unique at the connection scope.
-- [browser#992](https://github.com/grafana/xk6-browser/pull/992) Fixes go mod prometheus dependency issue.
```

</details>

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:51]: I'm working on a proposal for the changes, bear with me.

Comment (codebien) [reply release notes/v0.46.0.md:51]: I have the same feeling as @ankur22. It isn't introducing something new but just changing the API interface. If we have also a new part of the API then I'm fine with splitting them between breaking change and features sections but what is available right now in the text seems only a breaking change.
Of course, I don't have enough knowledge about the topic so I could be completely wrong, as I said this is mostly my feeling as a Reader.

File: release notes/v0.46.0.md
Comment (codebien) [reply release notes/v0.46.0.md:11]: I would suggest having the following structure:

## Breaking changes

### Automatic Browser Lifecycle Handling

A small introduction (as you already added in the current feature section)

- the list of breaking changes

It sounds a bit unfriendly to get directly the list of breaking changes without any introduction that provides context about so many breaking changes.

File: release notes/v0.46.0.md
Comment (inancgumus) [reply release notes/v0.46.0.md:146]: Consolidation makes more sense 👍

> would the log level changes from warn to debug be classed as improvements since the logs will be less noisy which will impact a user when they're debugging their test?

I think any UX change should be in the list as its own, and logs are one of them.

File: release notes/v0.46.0.md
Comment (inancgumus) [reply release notes/v0.46.0.md:11]: We can add a link to the migration guide here as well. More info: https://github.com/grafana/k6/pull/3260#discussion_r1290136033

Comment (ka3de) [reply release notes/v0.46.0.md]: See a proposal for reorganization in [9349278](https://github.com/grafana/k6/pull/3260/commits/93492783540421834b0fc09d1ebc1beda27e0499).

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md]: See a proposal for reorganization in [9349278](https://github.com/grafana/k6/pull/3260/commits/93492783540421834b0fc09d1ebc1beda27e0499), where this has been moved into the _breaking changes_ section.

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:144]: Coming back to this discussion, can you mention which one of the two diffs/presented options do you prefer or provide specific suggestions please? Thank you!

**Review by codebien — COMMENTED** [review]
LGTM, @ka3de can you rebase for resolving the conflicts, please?

File: release notes/v0.46.0.md
Comment (ankur22) [inline release notes/v0.46.0.md:18]: I think the tone needs to soften here a bit, how about something like: `**Metrics** also went through a few changes.`?

**Review by ankur22 — APPROVED** [review]
Nice! I like it 🙂

I've left one comment to consider, but otherwise it looks good.

File: release notes/v0.46.0.md
Comment (ka3de) [reply release notes/v0.46.0.md:18]: Right, I was not linking the word `suffer` but could not come up with a different word/expression at the moment :+1:

**Review by codebien — APPROVED** [review]

File: release notes/v0.46.0.md
Comment (ankur22) [reply release notes/v0.46.0.md:180]: I'm happy with the [second](https://github.com/grafana/k6/pull/3260#discussion_r1289901081) suggestion.

Comment (ka3de) [reply release notes/v0.46.0.md]: Fixed in [4cfd22b](https://github.com/grafana/k6/pull/3260/commits/4cfd22b02e55f31e31f5571be92892b87cf77ea1).

**Pull request merged**
